build.sh

```diff
@@ -15,19 +15,38 @@ echo "Installing dotnet CLI"
 mkdir -p cli
 curl -o cli/dotnet-install.sh -L https://dot.net/v1/dotnet-install.sh
 
+if [ $? -ne 0 ]; then
+	echo "Could not download 'dotnet-install.sh' script. Please check your network and try again!"
+	exit 1
+fi
+
 # Run install.sh for cli
 chmod +x cli/dotnet-install.sh
 
-# v1 needed for some test and bootstrapping testing version
+# Get recommended version for bootstrapping testing version
 cli/dotnet-install.sh -i cli -c 1.0
 
+if [ $? -ne 0 ]; then
+	echo ".NET CLI Install failed!!"
+	exit 1
+fi
+
 DOTNET="$(pwd)/cli/dotnet"
 
-echo "$DOTNET msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting"
+# Set IFS to new-line
+IFS=$'\n'
+
+# Let the dotnet cli expand and decompress first if it's a first-run
+$DOTNET --version
+
+# Get CLI Branches for testing
+echo "dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting"
+CMD_OUT_LINES=(`$DOTNET msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting`)
+DOTNET_BRANCHES=${CMD_OUT_LINES[-1]}
```

Comment:
what's the purpose of 45 and 46?
Why not assign to $DOTNET_BRANCH directly?

Reply:
This is the fix

Reply:
The next patch was not yet pushed at that time. It adds the version logic from `runFuncTests.sh`. I did this way to make diff between commits easy to understand visually.